src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int loaded;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.size = 0;
+        this.loaded = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeTable();
+        Node<K,V> newNode = new Node<>(keyHash(key), key, value, null);
+        Node<K,V> bucketToPut = table[getBucketIndexByKey(key)];
+        if (bucketToPut == null) {
+            table[getBucketIndexByKey(key)] = newNode;
+        } else {
+            while (bucketToPut != null) {
+                if (bucketToPut.hash == keyHash(key) && Objects.equals(bucketToPut.key, key)) {
+                    bucketToPut.value = value;
+                    return;
+                }
+                if (bucketToPut.next == null) {
+                    bucketToPut.next = newNode;
+                    break;
+                }
+                bucketToPut = bucketToPut.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> bucketToFind = table[getBucketIndexByKey(key)];
+        while (bucketToFind != null) {
+            if (bucketToFind.hash == keyHash(key) && Objects.equals(bucketToFind.key, key)) {
```

Comment:
Do we really need to check "bucketToFind.hash == keyHash(key) &&..." if we already get this bucket?